teuthology/suite/run.py

```diff
@@ -411,6 +410,7 @@ def matches(f):
 
             arg = copy.deepcopy(self.base_args)
             arg.extend([
+                '--num', str(self.args.num),
```

Comment:
how is `--num` connected with `write_result()` and `write_rerun_memo()` tasks?

Reply:
Directly via `build_base_args` as soon as it is has `--num X` in command line to `teuthology-schedule`, the latter actually put jobs in the beanstalk queue, the teuthology-schedule defaults --num to 1. So just removing this parameter will do the deal.

Reply:
my question still holds. i still don't understand how they are connected.

Reply:
Okay, to be precise it is connected not directly, but transitively.
Here are the beginning of `write_result()` a `write_rerun_memo()`:
```
    def write_rerun_memo(self):
        args = copy.deepcopy(self.base_args)
```
and
```
    def write_result(self):
        arg = copy.deepcopy(self.base_args)
```
both of them are deep coppying *self.base_args*, which is set in the beginning of `prepare_schedule()` using `sefl.build_base_args()`:
```
    def prepare_and_schedule(self):
        """
        Puts together some "base arguments" with which to execute
        teuthology-schedule for each job, then passes them and other parameters
        to schedule_suite(). Finally, schedules a "last-in-suite" job that
        sends an email to the specified address (if one is configured).
        """
        self.base_args = self.build_base_args()
```
The `self.build_base_args()` composes argument list, which includes `--num`. This patch, removes the `--num` from it on the line 297, so the base_args do not include it. 
I hope this helps to understand.